src/stores/media/media-item.ts

```diff
@@ -0,0 +1,81 @@
+import { defineStore } from 'pinia'
+
+import axios from 'axios'
+
+import type { Media } from '~/models/media'
+import type { SupportedMediaType } from '~/constants/media'
+import {
+  FetchState,
+  initialFetchState,
+  updateFetchState,
+} from '~/composables/use-fetch-state'
+import { services } from '~/stores/media/services'
+import { useMediaStore } from '~/stores/media/index'
+import { IMAGE } from '~/constants/media'
+
+export interface MediaItemState {
+  mediaItem: Media | null
+  mediaType: SupportedMediaType
+  fetchState: FetchState
+}
+
+export const useMediaItemStore = defineStore('media-item', {
+  state: (): MediaItemState => ({
+    mediaItem: null,
+    mediaType: IMAGE,
+    fetchState: initialFetchState,
+  }),
+  actions: {
+    _updateFetchState(action: 'end' | 'start', option?: string) {
+      this.fetchState = updateFetchState(this.fetchState, action, option)
+    },
+    /**
+     * Throws a new error with a new error message.
+     */
+    handleMediaError({ error }: { error: unknown }) {
+      let errorMessage
+      if (axios.isAxiosError(error)) {
+        errorMessage =
+          error.response?.status === 500
+            ? 'There was a problem with our servers'
+            : `Request failed with status ${
+                error.response?.status ?? 'unknown'
+              }`
+      } else {
+        errorMessage =
+          error instanceof Error ? error.message : 'Oops! Something went wrong'
+      }
+
+      this._updateFetchState('end', errorMessage)
+      throw new Error(errorMessage)
+    },
+
+    async fetchMediaItem(params: { type: SupportedMediaType; id: string }) {
+      const { type, id } = params
```

Comment:
I think it was a left over from previous implementation. Some time before it might have been a way to avoid problems with incorrect parameter order, or the mutation payload being an object most of the times. Now that there is just one parameter here, and TS suggesting which parameters function can take, it's not necessary.